src/contrib/mlir/lowerer.cpp

```diff
@@ -1029,6 +1007,27 @@ namespace
 
         rewriter.replaceOp(op, result);
     }
+
+    ValueHandle createZeroConstant(mlir::Type type)
+    {
+        if (auto floatTy = type.dyn_cast<FloatType>())
+        {
+            if (floatTy.isF32())
+                return intrinsics::constant_float(llvm::APFloat(0.0f), floatTy);
+            else if (floatTy.isF64())
+                return intrinsics::constant_float(llvm::APFloat(0.0), floatTy);
+            else
+                NGRAPH_CHECK(false, "Unsupported floating-point precision");
+        }
+        else if (auto intTy = type.dyn_cast<IntegerType>())
+        {
+            return intrinsics::constant_int(0, intTy.getWidth());
+        }
+        else
+        {
+            NGRAPH_CHECK(false, "Unsupported type");
+        }
+    }
```

Comment:
Nits:
- Please add { } for if-else statements based on coding standards.
- We can use NGRAPH_UNREACHABLE instead of NGRAPH_CHECK(false) 
- No need of `else` after `return`